### PR TITLE
Update links on flutter.dev/web page

### PR DESCRIPTION
**Pull request by kevmoo**
_Description of what this PR is changing or adding, and why:_

_Issues fixed by this PR (if any):_ Fixes #ISSUE-NUMBER

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/web/index.md [hunk 1]
```diff
@@ -88,7 +88,7 @@ The following resources can help you get started:
 [Flutter Gallery]: {{site.gallery}}
 [main Flutter repo]: {{site.repo.flutter}}
 [Preparing an app for web release]: {{site.url}}/deployment/web
-[Progressive Web Application]: {{site.developers}}/web/progressive-web-apps/
+[Progressive Web Application]: https://web.dev/progressive-web-apps/
```
Comment (domesticmouse) [inline src/web/index.md:91]: I suspect @sfshaza2 will want a var for `https://web.dev`

Comment (kevmoo) [reply src/web/index.md:91]: We don't have one for flutter.github.io 🤷

Comment (sfshaza2) [reply src/web/index.md:91]: I can live with this.  :). But, yeah, if we used it often, I'd want a macro.

Comment (sfshaza2) [reply src/web/index.md:91]: Too bad we don't have something more Flutter-specific to point to.

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Pull request merged**